.travis.yml

```diff
@@ -7,11 +7,17 @@ php:
  - 5.6
  - 7.0
 
+matrix:
+  fast_finish: true
+  include:
+    - php: 5.5
+      env: COMPOSER_FLAGS="--prefer-lowest"
+
 before_script:
  - bash -c 'if [ -n "$GH_TOKEN" ]; then composer config github-oauth.github.com ${GH_TOKEN}; fi;'
  - composer self-update
- - composer install --prefer-dist --no-interaction
  - composer require "squizlabs/php_codesniffer=*"
+ - travis_wait composer update --prefer-dist --no-interaction $COMPOSER_FLAGS
```

Comment:
Where does that statement stem from? IMHO this is just simply totally plain wrong.
If you test a range, you would need to test each and every version in every combination with every composer package you use.
Composer assumes that semver is used, hence if you carefully define your dependencies, no breaking change should occur, no matter which version is used.
The CI environment is not there to test the various dependency variations, but is there to validate that what you tested locally and committed is still valid, even if more/other changes to the code have been made.
Therefore it has to validate against the specific version you used locally. That's the reason why every package manager (composer, gem, ...) has a lock file.
That's also the sole reason why you commit the lock file. In order to have a reproducable state on CI systems.

If your composer library is required in a different project, your project's lock file is ignored anyway and the actually chosen lib versions are persisted in the other project's lock file, again documenting the exact versions of code that has been tested in this very project.

Check also https://getcomposer.org/doc/01-basic-usage.md#composer-lock-the-lock-file for more details.